sympy/solvers/ode.py

```diff
@@ -5890,6 +5890,9 @@ def _remove_coefficient(expr, x):
             exprs = tmpset
         return exprs
 
+    def is_homogeneous_solution(eq_homogeneous, func, act):
+            return any(expand(sub_func_doit(eq_homogeneous, func, ts)).is_zero for ts in act)
```

Comment:
```python
    def is_homogeneous_solution(term):
            return expand(sub_func_doit(eq_homogeneous, func, term)).is_zero
```

Reply:
That's what I meant by `is_homogeneous_solution`: a function that takes one expression and checks that it is a solution.